### PR TITLE
Update status for phantom uploading media.

### LINKED ISSUE #8310: Clear phantom `pushing` Media items on app startup

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #8310 

This PR by checking the status of Media when the app is started from scratch.
All media that is in the ```.pushing state``` is updated to ```.failed```.

Note that this behaviour will need to be changed when we start to do full async uploads, at that time we first need to check if there is no pending background task still going for that upload before deciding to change the status.

To test:
 - Start the app
 - Go to the media library
 - Add an media object and wait for start of upload of media
 - Kill the app (stop in Xcode)
 - Restart the app
 - Go to the same media library
 - See the status of media that was cancelled
 - You should be able to retry that media.

### CODE REVIEW

File: WordPress/Classes/Services/MediaCoordinator.swift [hunk 1]
```diff
@@ -214,6 +214,11 @@ class MediaCoordinator: NSObject {
         let service = MediaService(managedObjectContext: backgroundContext)
         service.syncMediaLibrary(for: blog, success: success, failure: failure)
     }
+
+    @objc func refreshMediaStatus() {
```
Comment (frosty) [inline WordPress/Classes/Services/MediaCoordinator.swift]: Could you add a doc comment to this, please?

Comment (frosty) [reply WordPress/Classes/Services/MediaCoordinator.swift]: And I guess to the corresponding service method.

File: WordPress/Classes/Services/MediaService.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+extension MediaService {
+
+    func refreshMediaStatus(onCompletion: (() -> Void)?, onError: ((Error) -> Void)?) {
```
Comment (frosty) [inline WordPress/Classes/Services/MediaService.swift]: If you set defaults for these parameters (`= nil`), then we can just omit them from the call site: `service.refreshMediaStatus()`.

File: WordPress/Classes/Services/MediaService.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                    DispatchQueue.main.async {
+                        onCompletion?()
+                    }
+                })
+
+            } catch {
+                DDLogError("Error while attempting to clean local media: \(error.localizedDescription)")
+                if let onError = onError {
+                    DispatchQueue.main.async {
+                        onError(error)
```
Comment (frosty) [inline WordPress/Classes/Services/MediaService.swift]: Can we just do `onError?(error)` here, like we do with `onCompletion`? (and remove the `if let`)

**Comment by SergioEstevao** [comment]
@frosty just addressed your comments, when you have time please give it another look.

File: WordPress/Classes/Services/MediaService.swift [hunk 4]
```diff
@@ -2,7 +2,14 @@ import Foundation
 
 extension MediaService {
 
-    func refreshMediaStatus(onCompletion: (() -> Void)?, onError: ((Error) -> Void)?) {
+    /// /// This method check the status of all media objects and updates them to the correct status if needed.
```
Comment (frosty) [inline WordPress/Classes/Services/MediaService.swift]: Nitpick – extra `///` here, and I'd change `check` to `checks` (that also applies to the comment in `MediaCoordinator` too) :)

File: WordPress/Classes/Services/MediaCoordinator.swift [hunk 5]
```diff
@@ -215,6 +215,9 @@ class MediaCoordinator: NSObject {
         service.syncMediaLibrary(for: blog, success: success, failure: failure)
     }
 
+    /// This method check the status of all media objects and updates them to the correct status if needed.
+    /// The main cause of wrong status is the app being killed while uploads of media are happening.
+    ///
     @objc func refreshMediaStatus() {
         let service = MediaService(managedObjectContext: backgroundContext)
         service.refreshMediaStatus(onCompletion: nil, onError: nil)
```
Comment (frosty) [inline WordPress/Classes/Services/MediaCoordinator.swift]: You can remove the parameters from this call now :) `service.refreshMediaStatus()`

**Comment by SergioEstevao** [comment]
@frosty ready for another go.

**Pull request merged**